tests/test_pkg.py

```diff
@@ -1,35 +1,107 @@
+"""Check the content of source code and test build source distribution.
+
+Binary distribution (wheel) is checked in test workflow.
+"""
+
 from __future__ import annotations
 
-import os
+import subprocess
+import tarfile
 from glob import glob
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from monty.tempfile import ScratchDir
+
+if TYPE_CHECKING:
+    from pymatgen.util.typing import PathLike
+
+PROJECT_ROOT = Path(__file__).parent.parent
+NAMESPACE_PKGS = {"analysis", "ext", "io"}  # TODO: double check
+
+
+def test_source_code():
```

Comment:
I'm not sure if that makes sense at all to compare source code/distribution with `SOURCES.txt` (would it be possible that the contents of sdist is defined by `SOURCES.txt`?), need to [double check the definition](https://setuptools.pypa.io/en/stable/deprecated/commands.html#egg-info-create-egg-metadata-and-set-build-tags):
> This command performs two operations: it updates a project’s .egg-info metadata directory (used by the bdist_egg, develop, and test commands), and it allows you to temporarily change a project’s version string, to support “daily builds” or “snapshot” releases. It is run automatically by the sdist, bdist_egg, develop, and test commands in order to update the project’s metadata, but you can also specify it explicitly in order to temporarily change the project’s version string while executing other commands. (It also generates the .egg-info/SOURCES.txt manifest file, which is used when you are building source distributions.)

In any case, I believe trying to test building sdist/bdist is necessary to make at least they could be built properly (bdist build is tested in test workflow)